otherlibs/unix/getgroups.c

```diff
@@ -28,11 +28,13 @@
 
 CAMLprim value caml_unix_getgroups(value unit)
 {
-  gid_t gidset[NGROUPS_MAX];
-  int n;
+  gid_t* gidset;
+  int n, groups_max;
   value res;
 
-  n = getgroups(NGROUPS_MAX, gidset);
+  groups_max = getgroups(0, NULL);
+  gidset = malloc(groups_max * sizeof(gid_t));
```

Comment:
I think this should be `caml_stat_alloc`, and also there should be a matching `(caml_stat_)free` call before `return res`, right?

Reply:
oops, fixed.